### DISCUSSION TITLE
Building a SSG service

### DISCUSSION THREAD

**Discussion by geordie2020** [vikejs/vike#930] [Discussion] [answered]
Hey guys,

I'm trying to use SSG from within an Express server. What i need is actually what `vite build` does, but programatically.
I've tried using `const { prerender } = require('vite-plugin-ssr/prerender')`, but as i understand it, this expects `vite build` to have run already to generate dist/server. Another problem with `prerender()` is, that it does not emit an assets directory, but solely .html files. 
What i'm trying to build is essentially a SSG service, which gets triggered by a get request along with some data to be rendered by the SSG and which writes the static site to disk (not sending it as response to the triggering request).
Could someone point me in the right direction?

**Comment by brillout** [comment, marked as answer]
You should be able to achieve that.

See https://vitejs.dev/guide/api-javascript.html#javascript-api. Also, FYI, there is a GitHub issue in this repo about someone doing something similar (revalidation a la Vercel ISR).

**Comment by geordie2020** [comment]
Thank you!
Is it expected behaviour, that the `build()` function of the Vite JS API does the same as `vite build` (npm script) when using VPS?

I think this is the issue, you're referring to: https://github.com/brillout/vite-plugin-ssr/issues/713

**Reply by brillout** [reply]
IIRC https://vite-plugin-ssr.com/disableAutoFullBuild is set to `true` when running `build()`.
